source/connection-monitoring-and-pooling/tests/pool-checkout-maxConnecting-timeout.yml

```diff
@@ -0,0 +1,71 @@
+version: 1
+style: integration
+description: waiting on maxConnecting is limited by WaitQueueTimeoutMS
+runOn:
+  -
+    # required for blockConnection in fail point
+    minServerVersion: "4.4.0"
+failPoint:
+  configureFailPoint: failCommand
+  # high amount to ensure not interfered with by monitor checks.
+  mode: { times: 20 }
+  data:
+    failCommands: ["isMaster"]
+    closeConnection: false
+    blockConnection: true
+    blockTimeMS: 750
+poolOptions:
+  maxPoolSize: 10
+  waitQueueTimeoutMS: 50
+operations:
+  # start creating two connections simultaneously.
+  - name: start
+    target: thread1
+  - name: checkOut
+    thread: thread1
+  - name: start
+    target: thread2
+  - name: checkOut
+    thread: thread2
+  # wait for other two threads to start establishing
+  - name: waitForEvent
+    event: ConnectionCreated
+    count: 2
+  # start a third thread that will be blocked waiting for
+  # one of the other two to finish
+  - name: start
+    target: thread3
+  - name: checkOut
+    thread: thread3
+  - name: waitForEvent
+    event: ConnectionCheckOutFailed
+    count: 1
+  # rejoin thread3, should experience error
```

Comment:
There's a slight problem with this test. This `waitForThread` operation does correctly cause Python's test runner to raise a WaitQueueTimeoutError exception which matches the `error` field. However, this means that the final `waitForEvent` (for ConnectionReady) never runs and therefore ConnectionReady does not appear in `events`.

I think python is following the spec test readme's behavior. Should we remove the `waitForEvent` and the `ConnectionReady` events? Or we could remove the `waitForThread` (thread 3) and the `error` field since we already validate that a ConnectionCheckOutFailed is emitted.

Reply:
In the Rust runner we actually continue to execute operations after receiving any errors from `waitForThread` and then compare at the end, though it does seem that it was the intention of the readme that the main thread just stops executing once it gets an error. I think the easiest approach here is just to drop the stuff that comes after the `waitForThread`. This doesn't really reduce our coverage at all either.

Reply:
SGTM. This test now passes in Python.